Common/ImageSamplers/itkImageReducedFullSampler.hxx

```diff
@@ -0,0 +1,145 @@
+/*=========================================================================
+ *
+ *  Copyright UMC Utrecht and contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef __ImageReducedFullSampler_txx
+#define __ImageReducedFullSampler_txx
+
+#include "itkImageReducedFullSampler.h"
+
+#include "itkImageRegionConstIteratorWithIndex.h"
+
+namespace itk
+{
+
+/**
+ * ******************* GenerateData *******************
+ */
+
+template <class TInputImage>
+void
+ImageReducedFullSampler<TInputImage>::GenerateData(void)
+{
+
+  /** Get handles to the input image, output sample container, and the mask. */
+  InputImageConstPointer                     inputImage = this->GetInput();
+  typename ImageSampleContainerType::Pointer sampleContainer = this->GetOutput();
+  typename MaskType::ConstPointer            mask = this->GetMask();
+
+  /** Clear the container. */
+  sampleContainer->Initialize();
+
+  /** Set up a region iterator within the user specified image region. */
+  typedef ImageRegionConstIteratorWithIndex<InputImageType> InputImageIterator;
+  InputImageIndexType                                       index = this->GetCroppedInputImageRegion().GetIndex();
+  index[ReducedInputImageDimension] = 0;
+  InputImageSizeType size = this->GetCroppedInputImageRegion().GetSize();
+  size[ReducedInputImageDimension] = 1;
+  InputImageRegionType region;
+  region.SetIndex(index);
+  region.SetSize(size);
+  InputImageIterator iter(inputImage, region);
```

Comment:
This appears to be the essential part, compared to the "non-reduced" FullSampler.

Reply:
@MathiasPolfliet @mstaring @stefanklein Is it really necessary for the "reduced" sampler to do `index[ReducedInputImageDimension] = 0`? Or would it be sufficient to only adjust the size, and leave the index alone?

I'm asking, because ITK allows specifying start index of the buffered region of an image to be non-zero. So an index value of zero may be outside the buffered region. Trying to retrieve samples outside the buffered region might cause crashes. So if it is sufficient for the "reduced" sampler to only adjust the size, I would prefer to leave the index alone. 

Another option might be something like:

    // Instead of index[ReducedInputImageDimension] = 0
    index[ReducedInputImageDimension] = bufferedRegion.GetIndex(ReducedInputImageDimension);

But I wonder if that's useful/necessary.